src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -0,0 +1,90 @@
+import { parse } from 'url';
+import type { Provider } from 'oidc-provider';
+// This import probably looks very hacky and it is. Weak Cache is required to get the oidc
+// configuration, which, in turn, is needed to get the routes the provider is using.
+// It is probably very difficult to get the configuration because Panva does not want
+// it to be possible, but we must get the configuration to satisfy the needs of the CSS
+// architecture. See the "canHandle" method for an explantaion
+// eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import instance from 'oidc-provider/lib/helpers/weak_cache';
```

Comment:
I don't mind the hackiness, but I do mind that this is likely to break in the future.

What are alternative solutions?

Reply:
The alternative is making the HttpHandler aware of the underlying configuration that was passed to IdentityProvider. And even then, we can't be 100% sure that that is the configuration that it's using since IdentityProvider has its own defaults.

Alternatively, we could ask Panva to make an official method that allows us to extract the config the provider is using.

Reply:
I'm fine with passing it in; Components.js can do this for us.

But even then, I want to know why this is needed.
So can we make this very explicit and specific? What pieces of information from the config are used and how?

Example: "the config has the string XYZ in place ABC, which we need to map the URL /xyz".

Reply:
Okay did it.